plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -153,29 +156,38 @@ public static CachingHiveMetastore memoizeMetastore(HiveMetastore delegate, long
         private HiveMetastore delegate;
         private Optional<Executor> executor = Optional.empty();
         private OptionalLong expiresAfterWriteMillis = OptionalLong.empty();
+        private OptionalLong statsExpiresAfterWriteMillis = OptionalLong.empty();
         private OptionalLong refreshMills = OptionalLong.empty();
         private Long maximumSize;
         private StatsRecording statsRecording = StatsRecording.ENABLED;
         private boolean partitionCacheEnabled = true;
+        private boolean metadataCacheEnabled = true;
+        private boolean statsCacheEnabled = true;
```

Comment:
Having defaults here looks error-prone, especially that these defaults

- are not same as the production defaults
- and not kept in sync with  them

let's make this required fields

1. declare them as `Boolean` here, initializing to null
2. in the build method wrap them like `requireNonNull(metadataCacheEnabled, "required metadataCacheEnabled not set")`